### PR TITLE
ignore files in .genignore before walking/substitution

### LINKED ISSUE #236: {{vars}} in files listed by .genignore cause a panic
If any ignored files have `{{this_style}}` vars, `cargo generate` errs during template substitution. 

Fixed by #235  

Example:
```bash
schell/m/s/lemon$ cargo generate --verbose --git https://github.com/schell/mogwai-template.git --name gen-test --branch master
 Creating project called `gen-test`...
Error:  Error replacing placeholders `/media/schell/lemon/gen-test/.github/workflows/cicd.yml`
Info: caused by liquid: Unknown variable
  with:
    requested variable=runner
    available variables=authors, crate_name, project-name
```

### PR DESCRIPTION
**Pull request by schell**
### changes
* ignores files listed in .genignore _before_ walking the project template and doing substitution 

### tl;dr
This should fix scenarios where liquid variables are causing errors in files that are ignored. Fixes #236

### CODE REVIEW

**Comment by schell** [comment]
@ashleygwilliams any thoughts on this?

**Comment by schell** [comment]
@EverlastingBugstopper thanks for the review - I'll add a test ASAP.

**Comment by schell** [comment]
I've added this test:
```rust
#[test]
fn it_removes_genignore_files_before_substitution() {
    let template = dir("template")
        .file(
            "Cargo.toml",
            r#"[package]
name = "{{project-name}}"
description = "A wonderful project"
version = "0.1.0"
"#,
        )
        .file(".cicd_workflow", "i contain a ${{ github }} var")
        .file(".genignore", r#".cicd_workflow"#)
        .init_git()
        .build();

    let dir = dir("main").build();

    Command::main_binary()
        .unwrap()
        .arg("gen")
        .arg("--git")
        .arg(template.path())
        .arg("-n")
        .arg("foobar-project")
        .current_dir(&dir.path())
        .assert()
        .success()
        .stdout(predicates::str::contains("Done!").from_utf8());

    assert_eq!(dir.exists("foobar-project/.cicd_workflow"), false);
}
```
which fails on `master` with:
```
---- basics::it_removes_genignore_files_before_substitution stdout ----
thread 'basics::it_removes_genignore_files_before_substitution' panicked at 'Unexpected failure.
code-1
stderr=```Error:  Error replacing placeholders `/media/schell/lemon/cargo-generate/target/generated-tests/test-18-main/foobar-project/.cicd_workflow`
Info: caused by liquid: Unknown variable
  with:
    requested variable=github
    available variables=authors, crate_name, project-name

...
```

The test passes on this PR's branch.
